modules/openapi-generator/src/main/resources/python/model_anyof.mustache

```diff
@@ -23,7 +23,10 @@ class {{classname}}({{#parent}}{{{.}}}{{/parent}}{{^parent}}BaseModel{{/parent}}
     # data type: {{{dataType}}}
     {{vendorExtensions.x-py-name}}: {{{vendorExtensions.x-py-typing}}}
 {{/composedSchemas.anyOf}}
-    actual_instance: Any
+    if TYPE_CHECKING:
+        actual_instance: Union[{{#anyOf}}{{{.}}}{{^-last}}, {{/-last}}{{/anyOf}}]
+    else:
+        actual_instance: Any
```

Comment:
@fa0311 thank again for the PR. Just a second thought, shall we simply use the following without TYPE_CHECKING:
```
    actual_instance: Union[{{#anyOf}}{{{.}}}{{^-last}}, {{/-last}}{{/anyOf}}]
```

Reply:
If TYPE_CHECKING is not used, an incorrect type conversion will be performed by pydantic.


```yaml
openapi: 3.0.3
info:
  title: Swagger Petstore
  version: 1.0.0

paths:
  /foo:
    get:
      responses:
        default:
          description: response
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Animal"

components:
  schemas:
    Animal:
      oneOf:
        - $ref: "#/components/schemas/Cat"
        - $ref: "#/components/schemas/Dog"
      discriminator:
        propertyName: type
        mapping":
          Cat: "#/components/schemas/Cat"
          Dog: "#/components/schemas/Dog"

    Cat:
      properties:
        type:
          type: string
        color:
          type: string

    Dog:
      properties:
        type:
          type: string
        bark:
          type: string
```

```yaml
inputSpec: test.yaml
outputDir: .
packageName: petstore
projectName: petstore
useOneOfDiscriminatorLookup: true
```

```python
from petstore import Animal
res = Animal.from_dict({"type": "Dog", "color": "brown"})
print(type(res.actual_instance)) # <class 'petstore.models.cat.Cat'>
```

In this example, the type is Dog, but the actual_instance is `<class 'petstore.models.cat.Cat'>`.
Assumed type of actual_instance is <class 'petstore.models.cat.Dog'>